scalafix-core/shared/src/main/scala/scalafix/internal/rule/ImplicitClassPrivateParams.scala

```diff
@@ -0,0 +1,25 @@
+package scalafix.internal.rule
+
+import scala.meta.quasiquotes.XtensionQuasiquoteInit
+import scala.meta.{Ctor, Defn, Mod, Template, Term}
+import scalafix.rule.RuleCtx
+import scalafix.{Patch, Rule}
+
+case object ImplicitClassPrivateParams
+    extends Rule("ImplicitClassPrivateParams") {
+
+  override def description: String =
+    "Add private access modifier to val or var parameters of implicit classes in order to prevent public access"
+
+  override def fix(ctx: RuleCtx): Patch = {
+    ctx.tree.collect {
+      case Defn.Class(cMods, _, _, Ctor.Primary(_, _, (Term.Param(pMods, _, _, _) :: Nil) :: Nil), Template(_, init"AnyVal" :: Nil, _, _)) =>
+        val optPatch = for {
+          _ <- cMods.find(_.is[Mod.Implicit])
+          if !pMods.exists(_.is[Mod.Private])
+          valMod <- pMods.find(_.is[Mod.ValParam])
```

Comment:
You are correct that this does not handle `final val param: T`. We can use `.find(!_.is[Mod.Annot])` instead, but adding private can still break cases like `override val param: T`. We should probably exclude cases that contain `override`.

Reply:
I think `override` isn't an issue since we're only looking for implicit value classes and it can't extend both `AnyVal` and a base class.

Reply:
AnyVals can inherit from "universal traits"

```scala
@ trait Foo extends Any {
    def bar: Int
    def plusOne = bar + 1
  }
defined trait Foo

@ implicit class UInt(override val bar: Int) extends AnyVal with Foo
defined class UInt

@ 2.plusOne
res4: Int = 3
```

See http://docs.scala-lang.org/overviews/core/value-classes.html

Reply:
Indeed. This case won't be taken into account though in your implem because you explicitly target a sole `Init` (`AnyVal` here). Should we remove this limitation and account for this : 
```scala
trait BaseClass {
  def bar: Int
}

implicit class XtensionOverrideVal(val bar: Int) extends AnyVal with BaseClass
```

Reply:
Indeed. This case won't be taken into account though, because in your implem you explicitly target a sole Init (AnyVal here).